core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1676,6 +1693,50 @@ public void _run() throws IOException, InstallationStatus {
             } else {
                 throw new SuccessButRequiresRestart(Messages._UpdateCenter_DownloadButNotActivated());
             }
+            } finally {
+                synchronized(this) {
+                    // There may be other threads waiting on completion
+                    LOGGER.fine("Install complete for: " + plugin.getDisplayName() + "@" + plugin.version);
+                    // some status other than Installing or Downloading needs to be set here
+                    // {@link #isAlreadyInstalling()}, it will be overwritten by {@link DownloadJob#run()}
+                    status = new Skipped();
+                    notifyAll();
+                }
+            }
+        }
+
+        /**
+         * Indicates there is another installation job for this plugin
+         * @since TODO
+         */
+        protected boolean isAlreadyInstalling() {
```

Comment:
I should probably rename this method... something like `wasInstalled`, will do so before merging